src/network/core/tcp_game.h

```diff
@@ -162,10 +165,17 @@ class NetworkGameSocketHandler : public NetworkTCPSocketHandler {
 
 	/**
 	 * Try to join the server:
-	 * string  OpenTTD revision (norev000 if no revision).
-	 * string  Name of the client (max NETWORK_NAME_LENGTH).
-	 * uint8_t   ID of the company to play as (1..MAX_COMPANIES).
-	 * uint8_t   ID of the clients Language.
+	 * string   OpenTTD revision (norev000 if no revision).
+	 * uint32_t NewGRF version.
+	 *
+	 * Before 15 there used to be more fields:
+	 * string   OpenTTD revision (norev000 if no revision).
+	 * uint32_t NewGRF version (added in 1.2).
+	 * string   Name of the client (max NETWORK_NAME_LENGTH).
+	 * uint8_t  ID of the company to play as (1..MAX_COMPANIES).
+	 * uint8_t  ID of the clients Language (functionally always set to 0).
+	 * string   Client's unique identifier (removed in 1.0).
```

Comment:
I added this documentation because for all intents and purposes `CLIENT_JOIN`, `SERVER_FULL`, `SERVER_BANNED`, `SERVER_ERROR` are supposed to be stable. Adding this documentation shows that `CLIENT_JOIN` is not fully stable, even though earlier documentation (PacketType) suggests that it must be.

There is also `CLIENT_GAME_INFO` and `SERVER_GAME_INFO`, though they have internal versioning and all versions written out.

Reply:
GameInfo is different, as in they have all versions implemented in the code too. This is just a comment that can also be found back in git history. So I am still trying to find a reason to keep this here. As in, what purpose it serves to keep it around, other than creating confusion as there is no code dealing with the pre-15 information.

Compare it with `PACKET_CLIENT_UNUSED`, which also no longer contains any details how the packet looked.

If you want to keep it, maybe do it a bit in reverse:

```
	 * string   OpenTTD revision (norev000 if no revision).
	 * uint32_t NewGRF version (added in 1.2).
	 * string   Name of the client (max NETWORK_NAME_LENGTH) (removed in 15).
	 * uint8_t  ID of the company to play as (1..MAX_COMPANIES) (removed in 15).
	 * uint8_t  ID of the clients Language (removed in 15).
	 * string   Client's unique identifier (removed in 15).
```

That also highlights better that the first two fields are kept around.